extension/packagemanagerextension/nbextension/methods.js

```diff
@@ -0,0 +1,148 @@
+            let updates = resp.updates;
+            let packages = [];
+            let ct = 0;
+
+            if (selectedPackages.length > 0) {
+                html += "<ul>";
+                for (let element of selectedPackages) {
+                    element = element.split('=');
+                    let pkg = element[0];
+                    let ver = element[1];
+                    let nver = check_new_version(updates, pkg, ver);
+                    if (nver != ver) {
+                        packages.push(pkg);
+                        html += "<li>";
+                        html += pkg + "    " + ver + " -> " + nver;
+                        html += "</li>";
+                        ct += 1;
+                    }
+                }
+                html += "</ul>";
+            }
+            else {
+                let list = $('.installed-values');
+                html += "<ul>";
+                for (let item of list) {
+                    let pkg = $(item).find('.value-name')[0].innerText;
+                    let ver = $(item).find('.value-version')[0].innerText;
+                    let nver = check_new_version(updates, pkg, ver);
+                    if (nver != ver) {
+                        packages.push(pkg);
+                        html += "<li>";
+                        html += pkg + "    " + ver + " -> " + nver;
+                        html += "</li>";
+                        ct += 1;
+                    }
+                }
+                html += "</ul>";
+            }
+            if (ct === 0) {
+                html = "<p> There are no updates available </p>"
+                common.dispay_msg("Install Packages", $.parseHTML(html));
+            }
+            else
+                common.confirm("Update Packages", $.parseHTML(html), "Confirm", function () {
+                    api.update_packages(packages, project);
+                });
+        });
+    }
+
+
+    /*
+    This function lists down all the packages selected for removal and requires a user confirmation.
+    */
+
+    function delete_packages(project) {
+        $('#deletebtn').unbind();
+        $('#deletebtn').click(function () {
+            let selectedPackages = common.get_selected_packages();
+            let html = "<p> The following packages are about to be deleted: </p> </br>";
+            html += "<ul>";
+            let packages = [];
+            for (let element of selectedPackages) {
+                element = element.split('=')[0];
+                packages.push(element);
+                html += "<li>";
+                html += element;
+                html += "</li>";
+            }
+            html += "</ul>";
+            if (packages.length === 0) {
+                html = "<p> No packages selected </p>";
+                common.dispay_msg("Install Packages", $.parseHTML(html));
+            }
+            else
+                common.confirm("Delete Packages", $.parseHTML(html), "Confirm", function () {
```

Comment:
I also told you - in a previous evaluation - that if you have a lot of times that you need to create a list, convert to html, and then put that in the popups, why don't you build the html of the list inside the popups? Then this block of code (and the previous ones) would be so much smaller...